ligo/types.mligo

```diff
@@ -7,6 +7,82 @@
 
 (* Tick types, representing pieces of the curve offered between different tick segments. *)
 type tick_index = {i : int}
+
+(* FA2 related types *)
+
+type position_id = nat
+
+type operator =
+  { owner : address
+  ; operator : address
+  }
+type operators = (operator, unit) big_map
+
+type ledger_key = address * position_id
```

Comment:
@diogo Make sense, I didn't think of this. 

One question that comes up with this is that, we have `position_map` that is indexed by `position_id` as well, should we combine this with `ledger` (by adding `owner`, `bal`, `lo` and `hi` as fields to `position_state`) or we rather keep them separate.

Reply:
Now that I think about this, we probably does not need `ledger` at all, I'll update this in a moment.

Reply:
On another note, I want to clarify something:

> We need to start tracking positions by a nat id, one option might be to replace the current:
> `type position_map = (position_index, position_state) big_map`
> with something like two:
> ```
> type position_map = (position_id, position_state) big_map
> type position_id_map = (position_index, position_id) big_map .
> ```
> in such a way, we'd still be able to access a position by only knowing its position_index, but > we'll also be able to switch ownership of a position without updating both big_map.

I think there might be some confusion since, the owner is stored in `position_index` not the `position_state` so we have to update both maps.

Since we are going to update both maps anyway, I propose using this instead:

```
type position_map = (position_index, position_state) big_map
type position_index_map = (position_id, position_index) big_map .
```

With this, we don't have to use the `ledger` map (simply update both `position_index` with the new owner). What do you think about this? cc @Martoon-00